src/installer/installer.go

```diff
@@ -243,8 +244,72 @@ func (i *installer) updateSingleNodeIgnition(singleNodeIgnitionPath string) erro
 	return nil
 }
 
-func (i *installer) writeImageToDisk(ignitionPath string) error {
-	i.UpdateHostInstallProgress(models.HostStageWritingImageToDisk, "")
+func convertToInstallerKargs(kargs []string) []string {
+	var ret []string
+	for _, karg := range kargs {
+		ret = append(ret, "--append-karg", karg)
+	}
+	return ret
+}
+
+func convertToOverwriteKargs(args []string) []string {
+	var ret []string
+	i := 0
+	for i < len(args) {
+		if args[i] == "--append-karg" && i+1 < len(args) {
+			ret = append(ret, "--karg", args[i+1])
+			i += 2
+		} else {
+			i++
+		}
+	}
+	return ret
+}
+
+func (i *installer) getEncapsulatedMC(ctx context.Context, ignitionPath string, mc **mcfgv1.MachineConfig) func() error {
+	return func() error {
+		interval := 5 * time.Second
+		var err error
+		_ = utils.WaitForPredicateWithContext(ctx, 2*time.Hour, interval, func() bool {
+			*mc, err = i.ops.GetEncapsulatedMC(ignitionPath)
+			if err != nil {
+				i.log.WithError(err).Error("failed to get encapsulated MC")
+			}
+			return err == nil
+		})
+		if err != nil {
+			return fmt.Errorf("failed getting required kargs: %w", err)
+		}
```

Comment:
Unclear how this is related to the rest of the function

Reply:
The error is what received from the function GetEncapsulatedMC. The function WaitForPredicateWithContext does not return it.  This is the error that actually interests us.